frontend/src/components/dashboard/Analyzer.tsx

```diff
@@ -166,111 +219,166 @@ const Analyzer = ({ session }: { session: ExistingSessionExtended }) => {
-              splits={splits}
-              setSplits={setSplits}
-            />
-          </Col>
-          <Col xs={12}>
-            {task && (
-              <CodeComparison
-                classId={session.klass.id}
-                sessionId={session.id}
-                taskId={task.id}
-                subTaskId={selectedSubTaskId}
+                {isAutomaticGrouping && (
+                  <Input
+                    label={messages.numberOfGroups}
+                    type="number"
+                    value={numberOfGroups}
+                    onChange={(e) =>
+                      setNumberOfGroups(parseInt(e.target.value))
+                    }
+                  />
+                )}
+
+                {!isGroupingAvailable && (
+                  <FormattedMessage
+                    id="Analyzer.noGroupingAvailable"
+                    defaultMessage="Computing groups, please be patient."
+                  />
+                )}
+              </Parameters>
+            </Col>
+            <Col xs={12} lg={9}>
+              <Analysis
                 taskType={task.type}
-                groupAssignments={groupAssignments}
-                groups={groups}
+                xAxis={xAxis}
+                setXAxis={updateXAxis}
+                yAxis={yAxis}
+                setYAxis={updateYAxis}
+                categorizedDataPoints={categorizedDataPoints}
+                manualGroups={manualGroups}
+                splittingEnabled={!isAutomaticGrouping}
+                splits={splits}
+                setSplits={setSplits}
+                onSelectSolution={onSelectSolution}
               />
-            )}
-          </Col>
-        </Row>
-      )}
-    </MultiSwrContent>
+            </Col>
+            <Col xs={12}>
+              {task && (
+                <CodeComparison
+                  classId={session.klass.id}
+                  sessionId={session.id}
+                  taskId={task.id}
+                  subTaskId={selectedSubTaskId}
+                  taskType={task.type}
+                  groupAssignments={groupAssignments}
+                  groups={groups}
+                  selectedLeftGroup={selectedLeftGroup}
+                  setSelectedLeftGroup={setSelectedLeftGroup}
+                  selectedRightGroup={selectedRightGroup}
+                  setSelectedRightGroup={setSelectedRightGroup}
+                  selectedLeftSolution={selectedLeftSolution}
+                  setSelectedLeftSolution={setSelectedLeftSolution}
+                  selectedRightSolution={selectedRightSolution}
+                  setSelectedRightSolution={setSelectedRightSolution}
+                />
+              )}
+            </Col>
+          </Row>
+        )}
+      </MultiSwrContent>
+      <Modal
+        show={selectedSolutionId !== undefined}
+        onHide={() => setSelectedSolutionId(undefined)}
+        data-testid="solution-selection-modal"
+      >
```

Comment:
Somehow when I click on a student, I'd almost expect to be able to click "left" or "right" there and then (with minimal mouse movement), whereas this modal appears much further up in the window. It doesn't feel natural.
Could we perhaps have this modal more centered on the graph and/or somehow closer to where the mouse (and user attention) is ? 😅

Reply:
Discussed in person - not right now